server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -1857,6 +1876,8 @@ public static IndexMetadata legacyFromXContent(XContentParser parser) throws IOE
             }
             XContentParserUtils.ensureExpectedToken(XContentParser.Token.END_OBJECT, parser.nextToken(), parser);
 
+            builder.putMapping("{\"properties\": {}}"); // just make sure it's not empty so that _source can be read
```

Comment:
Can we use something like `builder.putMapping(MappingMetadata.EMPTY_MAPPINGS);` instead?

Reply:
++ nice